### PR TITLE
Fix only SHA1 checksum is accepted when uploading resources

### PR DESCRIPTION
**Pull request by tboye**
Depending on the storage type and settings, it might happen that resource metadata will provide a non-SHA1 checksum. At the moment it causes an exception. This MR fixes that and makes resource uploader accept all other supported checksum types.

In addition, a similar inconsistency is fixed within CSV export implementation.

### CODE REVIEW

File: udata/core/dataset/api.py [hunk 1]
```diff
@@ -362,7 +363,10 @@ def handle_upload(self, dataset):
         if 'html' in infos['mime']:
             api.abort(415, 'Incorrect file content type: HTML')
         infos['title'] = os.path.basename(infos['filename'])
-        infos['checksum'] = Checksum(type='sha1', value=infos.pop('sha1'))
+        for checksum_type in CHECKSUM_TYPES:
+            if checksum_type in infos:
+                infos['checksum'] = Checksum(type=checksum_type, value=infos.pop(checksum_type))
+                break
```
Comment (maudetes) [inline udata/core/dataset/api.py:369]: `infos['checksum']` may never be set if none of the checksum_type matches and it wouldn't raise an error here.
Using `next()` would raise an error directly instead, what do you think?

```suggestion
        checksum_type = next(algo for algo in CHECKSUM_TYPES if algo in infos)
        infos['checksum'] = Checksum(type=checksum_type, value=infos.pop(checksum_type))
```

**Review by maudetes — COMMENTED** [review]
Thank you a lot for this contribution! :pray:
I'm making some suggestions here.

I'm also wondering if we could add some relevant tests, but not sure we have fixtures to test a storage returning a checksum other than `sha1` yet.

File: udata/core/dataset/api.py (hunk 1 above)
Comment (tboye) [reply udata/core/dataset/api.py:369]: I was actually hesitating whether we want to raise an exception here or wait for the caller to complain, but after all I agree it's best to do it here. Also, looks way more elegant with `next()`.

**Comment by tboye** [comment]
@maudetes Regarding testing, from what I see there's indeed no infrastructure for that at the moment. Do you have any suggestions on how to address that?

**Review by maudetes — APPROVED** [review]
Thank you for the update!

Regarding the tests, I think a StorageMock could be created in `test_storages`.
But regarding the new code lines, it may be overkill and I'm okay with merging it without additional tests (`sha1` case is already tested).

**Comment by maudetes** [comment]
> Just to make sure, do changes in `udata/core/dataset/tasks.py` also make sense? It was implemented in a bit strange way so I'm wondering if there was some reasoning behind it.

I also found the logic weird, but I think it was erroneous, taking a look at the rest of the logic.
All tests pass on the csv export functionality, and it runs correctly locally with a `sha1` storage (ex: running `udata job run export-csv organization`).

**Comment by maudetes** [comment]
Well done for this work! I can merge if you update the branch with master :blush:

**Pull request merged**